public/content/lessons/2017/chain-rule-and-product-rule/index.mdx

```diff
@@ -7,4 +7,355 @@ video: YG15m2VwSjA
+
+<Figure
+  image="figures/chain-rule-dh.svg"
+  caption="The fact that $\sin(h)$ is moving left while the $dh$ bump is to the right just means that this change $d\left(\sin(h)\right)$ is some negative number."
+/>
+
+Now think of that third value, which is pegged at $\sin(h)$. It’s change $d\left(\sin(h)\right)$; the tiny change *caused* by the nudge $dh$. Well, we know the derivative of sine, so we can expand $d\left(\sin(h)\right)$ as $\cos(h) \cdot dh$; that’s what it means for the derivative of sine to be cosine.
+
+<Figure
+  image="figures/chain-rule-example-dh-derivative.svg"
+  caption="The derivative of $f(x) = \sin(x^2)$ is reframed by using the transformation $h=x^2$."
+/>
+
+Now we can unfold the transformation, replacing $h$ with $x^2$ and $dh$ with $d(x^2)$. So the bottom nudge becomes $\cos(x^2)d(x^2)$ and the middle nudge becomes $d(x^2)$. Of course, we also know that $d(x^2) = 2x \cdot dx$ and so we can substitute that into the diagram as well.
+
+<Figure
+  image="figures/chain-rule-example-dh-expanded.svg"
+/>
+
+It’s always good to remind yourself of what this all actually means. In this case where we started at $x = 1.5$ up top, this means that the size of that nudge on the third line is about $\cos(1.5^2) \cdot 2(1.5)$ multiplied by the size of $dx$; proportional to the size of $dx$, where the derivative here gives us that proportionality constant.
+
+Since the nudge on the third line represents the change to our initial function $df$ when we introduced the small nudge $dx$, we can rearrange the expression and this gives us the derivative of the function.
+
+<Figure
+  image="figures/chain-rule-example-derivative.svg" 
+/>
+
+Notice what we have here, we have the derivative of the outside function, still taking in the unaltered inside function, and we multiply it by the derivative of the inside function.
+
+<Figure
+  image="figures/chain-rule-annotated.svg"
+/>
+
+Again, there’s nothing special about $\sin(x)$ and $x^2$. If you have two functions $g(x)$ and $h(x)$, the derivative of their composition function $g\left(h(x)\right)$ is the derivative of $g$, evaluated at $h(x)$, times the derivative of $h$. This is what we call the “chain rule”.
+
+<Figure
+  image="figures/chain-rule-general-annotated-alt.svg"
+  caption="The chain rule for the function $f(x) = g(h(x))$"
+/>
+
+Notice, for the derivative of $g$, I’m writing it as $\frac{dg}{dh}$ instead of $\frac{dg}{dx}$. On the symbolic level, this serves as a reminder that you still plug in the inner function to this derivative. But it’s also an important reflection of what this derivative of the outer function actually represents.
```

Comment:
The text here says dg/dh and dg/dx, but in the image the notation used is d/dh[g]. I think that difference could be confusing for someone new to calculus. I think it's probably best to just change the text to match the image notation.